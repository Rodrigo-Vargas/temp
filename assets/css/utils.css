.opened-nav .top-nav {
  right: 0;
}

.opened-nav .btn.nav-toggle {
  border-color: #fff;
  color: #fff;
}
