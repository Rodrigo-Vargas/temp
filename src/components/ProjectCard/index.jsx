import React from 'react';
import PropTypes from 'prop-types';

import {
  Wrapper, TagList, Tag, Title, Link,
} from './styles';

const ProjectCard = ({
  title, img, slug, index,
}) => (
  <Wrapper key={index}>
    <Link href={`/projects/${slug}`} data-testid="project-link">
      <TagList>
        <Tag>HTML</Tag>
        <Tag>Wordpress</Tag>
      </TagList>
      <Title>{title}</Title>
      <div className="absolute bottom-0">
        <div className="h-36 overflow-hidden w-4/5 mx-auto">
          <img src={img} alt={title} />
        </div>
      </div>
    </Link>
  </Wrapper>
);

ProjectCard.propTypes = {
  index: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
};

export default ProjectCard;
