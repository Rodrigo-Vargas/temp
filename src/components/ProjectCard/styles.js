import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding-left: ${theme.spacers.p2};
    padding-right: ${theme.spacers.p2};
  `}
  width: 100%;
`;

export const TagList = styled.div`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacers.p5};
    padding-top: ${theme.spacers.p2};
  `}
`;

export const Tag = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.gray300};
    padding-bottom: ${theme.spacers.p1};
    padding-right: ${theme.spacers.p5};
    padding-top: ${theme.spacers.p1};
  `}
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.gray800};
    font-size: ${theme.font.sizes['2xl']};
    margin-bottom: ${theme.spacers.p3};
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    min-height: 16rem;

    --tw-bg-opacity: 1;
    background-color: rgba(249, 250, 251, var(--tw-bg-opacity));
    border-radius: 0.375rem;
    display: block;
    margin-bottom: 0.75rem;
    padding: 1.25rem;
    padding-bottom: 12rem;
    position: relative;
    --tw-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
    --tw-translate-x: 0;
    --tw-translate-y: 0;
    --tw-rotate: 0;
    --tw-skew-x: 0;
    --tw-skew-y: 0;
    --tw-scale-x: 1;
    --tw-scale-y: 1;
    transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
    transition-property: all;
    transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
    transition-duration: 150ms;
    transition-duration: 300ms;
  `}
`;

export const Another = {};
