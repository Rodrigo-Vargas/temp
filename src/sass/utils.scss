.opened-nav {
   .top-nav {
      right: 0;
   }

   .btn.nav-toggle {
      border-color: #fff;
      color: #fff;      
   }
}
