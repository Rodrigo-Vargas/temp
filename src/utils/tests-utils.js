/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import { render } from '@testing-library/react';
import TransitionLinkProvider from 'gatsby-plugin-transition-link/context/InternalProvider';

import Theme from '../styles/theme';

const customRender = (ui, renderOptions) => render(
  <Theme><TransitionLinkProvider>{ui}</TransitionLinkProvider></Theme>, renderOptions,
);

export * from '@testing-library/react';
export { customRender as render };
