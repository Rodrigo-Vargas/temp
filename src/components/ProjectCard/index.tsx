import Image from 'next/image';
import * as Icons from 'components/Icons';

import {
  TagList,
  Tag,
  Title,
  LinkTitle,
  CardWrapper,
  ImageBox,
  ImageWrap,
  CardHeader,
  Meta,
  MetaLink,
} from './styles';

interface ProjectCardProps {
  categories: Array<string>;
  img: string;
  link: string;
  slug: string;
  title: string;
}

const ProjectCard: React.FC<ProjectCardProps> = ({
  categories,
  img,
  link,
  slug,
  title,
}) => {
  console.log(img);
  return(
    <CardWrapper>
      <CardHeader>
        <LinkTitle href={`/projects/${slug}`} data-testid="project-link">
          <Title>{title}</Title>
        </LinkTitle>
        <Meta>
          {link && (
            <MetaLink target="_blank" href={link}>
              <Icons.ExternalLink style={{ transform: 'scale(0.8)' }} />
            </MetaLink>
          )}
        </Meta>

        <TagList>
          {categories?.map((category, i) => (
            <Tag key={i}>{category}</Tag>
          ))}
        </TagList>
      </CardHeader>
      <ImageBox>
        <ImageWrap>
          {
            (img) &&
              (
                <img src={img} alt={title} />
              )
          }

        </ImageWrap>
      </ImageBox>
    </CardWrapper>
  )
};

export type ProjectCard = {
  index: number,
  title: string,
  img: string,
  slug: string,
  link: string,
};

export default ProjectCard;
