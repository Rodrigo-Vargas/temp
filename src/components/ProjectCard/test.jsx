import React from 'react';
import ProjectCard from '.';

import { render, screen } from '../../utils/tests-utils';

const props = {
  id: '1',
  title: 'Sample Post',
  img: 'https://source.unsplash.com/user/willianjusten/300x140',
  slug: 'sample-post',
  index: 1,
};

describe('<ProjectCard />', () => {
  it('should render correclty', () => {
    const { container } = render(<ProjectCard {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.img,
    );

    expect(screen.getByTestId('project-link', { name: props.title })).toHaveAttribute(
      'href',
      `/projects/${props.slug}`,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
