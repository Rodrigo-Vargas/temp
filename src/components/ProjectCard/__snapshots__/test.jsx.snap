// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard /> should render correclty 1`] = `
<div
  class="col w-full sm:w-1/2"
>
  <a
    class="card featured"
    data-testid="project-link"
    href="/projects/sample-post"
  >
    <div
      class="py-5 pt-2"
    >
      <span
        class="text-gray-300 text-sm pr-5 py-1"
      >
        HTML
      </span>
      <span
        class="text-gray-300 text-sm pr-5 py-1"
      >
        Wordpress
      </span>
    </div>
    <h2
      class="text-gray-800 text-2xl mb-3"
    >
      Sample Post
    </h2>
    <div
      class="absolute bottom-0"
    >
      <div
        class="h-36 overflow-hidden w-4/5 mx-auto"
      >
        <img
          alt="Sample Post"
          src="https://source.unsplash.com/user/willianjusten/300x140"
        />
      </div>
    </div>
  </a>
</div>
`;
